src/main/scala/com/actian/spark_vector/datastream/DataStreamClient.scala

```diff
@@ -13,36 +13,43 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.actian.spark_vector.writer
+package com.actian.spark_vector.datastream
 
 import java.sql.SQLException
+import java.sql.ResultSet
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 import org.apache.spark.Logging
 import com.actian.spark_vector.util.ResourceUtil.closeResourceOnFailure
 import com.actian.spark_vector.vector.{ VectorConnectionProperties, VectorJDBC }
 import com.actian.spark_vector.vector.VectorException
 import com.actian.spark_vector.vector.ErrorCodes
+import com.actian.spark_vector.BooleanExpr
 
 /**
  * A client to prepare loading and issue the load `SQL` query to Vector
- *
  * @param vectorProps connection information
  * @param table to which table this client will load data
  *
  * @note This client opens a JDBC connection when instantiated. To prevent leaks,
  *  the [[close]] method must be called
  */
-case class DataStreamClient(vectorProps: VectorConnectionProperties,
-    table: String) extends Serializable with Logging {
+case class DataStreamClient(vectorProps: VectorConnectionProperties, table: String) extends Serializable with Logging {
   private val jdbc = {
     val ret = new VectorJDBC(vectorProps)
     ret.autoCommit(false)
     ret
   }
 
-  private def startLoadSql(table: String) = s"copy table $table from external"
   private def prepareLoadSql(table: String) = s"prepare for x100 stream into $table"
+  private def startLoadSql(table: String) = s"copy table $table from external"
+
+  private def prepareUnloadSql(table: String) = s"prepare for x100 stream from $table"
+  private def startUnloadSql(selectQuery: String) = s"insert into external table $selectQuery"
+
+  private def executeSql(sql: String, whereParams: Seq[Any] = Seq.empty): Future[Int] = Future {
+    whereParams.isEmpty.ifThenElse(jdbc.executeStatement(sql), jdbc.executePreparedStatement(sql, whereParams))
```

Comment:
the ifThenElse doesn't seem to reduce the amount of code, does it?
if (whereParams.isEmpty) jdbc.executeStatement(sql) else jdbc.executePreparedStatement(sql, whereParams)